xl-v1.go

```diff
@@ -144,21 +144,27 @@ func newXLObjects(disks, ignoredDisks []string) (ObjectLayer, error) {
 	}
 
 	// Handles different cases properly.
-	switch reduceFormatErrs(sErrs, len(storageDisks)) {
-	case errUnformattedDisk:
-		// All drives online but fresh, initialize format.
-		if err := initFormatXL(storageDisks); err != nil {
-			return nil, fmt.Errorf("Unable to initialize format, %s", err)
+	for _, err := range reduceFormatErrs(sErrs, len(storageDisks)) {
+		switch err {
```

Comment:
Is the logic here to run across all errors and write 'format.json' again and again?